discovery/chan_series.go

```diff
@@ -204,11 +205,50 @@ func (c *ChanSeries) FilterKnownChanIDs(_ chainhash.Hash,
 	isZombieChan func(time.Time, time.Time) bool) (
 	[]lnwire.ShortChannelID, error) {
 
-	newChanIDs, err := c.graph.FilterKnownChanIDs(superSet, isZombieChan)
+	newChanIDs, knownZombies, err := c.graph.FilterKnownChanIDs(superSet)
 	if err != nil {
 		return nil, err
 	}
 
+	edgeLookup := make(map[uint64]graphdb.ChannelUpdateInfo)
+	for _, edge := range superSet {
+		edgeLookup[edge.ShortChannelID.ToUint64()] = edge
+	}
+
+	for _, scid := range knownZombies {
+		info, ok := edgeLookup[scid]
+		if !ok {
+			return nil, fmt.Errorf("returned zombie channel was " +
+				"not in the original set of channel updates")
+		}
+
+		// TODO(ziggie): Make sure that for the strict pruning case we
+		// compare the pubkeys and whether the right timestamp is not
+		// older than the `ChannelPruneExpiry`.
+		//
+		// NOTE: The timestamp data has no verification attached to it
+		// in the `ReplyChannelRange` msg so we are trusting this data
+		// at this point. However it is not critical because we are
+		// just removing the channel from the db when the timestamps are
+		// more recent. During the querying of the gossip msg
+		// verification happens as usual. However we should start
+		// punishing peers when they don't provide us honest data ?
+		isStillZombie := isZombieChan(
```

Comment:
One thing I don't see here is an update in unit tests to target this shift. Before this change, from the PoV of the method (assuming it has no intimate knowledge about the referenced interface), the logic didn't affect any DB records, but it does now. 

I understand the motive to move some of the logic out from the channel graph, but perhaps an intermediate step would be to _extract_ those methods related to gossip queries into a new struct/interface, which would accomplish the same goal, but via a layer of indirection. 

Otherwise, you've eliminated responsibilities from one area, only to add some _entirely new_ responsibilities in another area.

Reply:
yes good point. 

The next set of PRs is going to separate out the graph cache from the ChannelGraph (ie, a new layer of abstraction there). So i think it will actually just be better to move this logic to that layer 👍